src/components/Tipping/index.tsx

```diff
@@ -0,0 +1,441 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import React, { useEffect, useState } from 'react';
+import { Alert, Button, Form, Input, Modal, Spin } from 'antd';
+import { useCurrentTokenDataSelector, useNetworkSelector, useUserDetailsSelector } from '~src/redux/selectors';
+import { useApiContext } from '~src/context';
+import { LoadingStatusType, NotificationStatus, Wallet } from '~src/types';
+import getAccountsFromWallet from '~src/util/getAccountsFromWallet';
+import { InjectedTypeWithCouncilBoolean } from '~src/ui-components/AddressDropdown';
+import { InjectedWindow } from '@polkadot/extension-inject/types';
+import AvailableWallets from '~src/ui-components/AvailableWallet';
+import AccountSelectionForm from '~src/ui-components/AccountSelectionForm';
+import BN from 'bn.js';
+import { poppins } from 'pages/_app';
+import BalanceInput from '~src/ui-components/BalanceInput';
+import styled from 'styled-components';
+import queueNotification from '~src/ui-components/QueueNotification';
+import executeTx from '~src/util/executeTx';
+import { inputToBn } from '~src/util/inputToBn';
+import nextApiClientFetch from '~src/util/nextApiClientFetch';
+import { MessageType } from '~src/auth/types';
+import getSubstrateAddress from '~src/util/getSubstrateAddress';
+import { formatedBalance } from '~src/util/formatedBalance';
+import { chainProperties } from '~src/global/networkConstants';
+import { formatBalance } from '@polkadot/util';
+import HelperTooltip from '~src/ui-components/HelperTooltip';
+
+import Tip1Icon from '~assets/icons/tip-1.svg';
+import Tip2Icon from '~assets/icons/tip-2.svg';
+import Tip3Icon from '~assets/icons/tip-3.svg';
+import Tip4Icon from '~assets/icons/tip-4.svg';
+import SaySomethingIcon from '~assets/icons/say-something.svg';
+import CloseIcon from '~assets/icons/close.svg';
+import TipIcon from '~assets/icons/tip-title.svg';
+
+const ZERO_BN = new BN(0);
+
+interface Props {
+	open: boolean;
+	setOpen: (pre: boolean) => void;
+	className?: string;
+	destinationAddress: string;
+	username: string;
+}
+
+const TIPS: { key: 'threeDollar' | 'fiveDollar' | 'tenDollar' | 'fifteenDollar'; value: number }[] = [
+	{ key: 'threeDollar', value: 3 },
+	{ key: 'fiveDollar', value: 5 },
+	{ key: 'tenDollar', value: 10 },
+	{ key: 'fifteenDollar', value: 15 }
+];
+
+const Tipping = ({ className, destinationAddress, open, setOpen, username }: Props) => {
+	const { network } = useNetworkSelector();
+	const { loginWallet, loginAddress } = useUserDetailsSelector();
+	const { currentTokenPrice } = useCurrentTokenDataSelector();
```

Comment:
if user directly visits some other page then this will break. Use a fallback and reset in redux